update user roles and permissions (#3537)
temporalio/documentation: docs/production-deployment/cloud/get-started/users.mdx

- **Global Admin**
  - Has full administrative permissions across the account, including users and usage

>   - Has Namespace Admin [permissions](#namespace-level-permissions) on all [Namespaces](/namespaces) in the account (can not be revoked)

Reviewer: nit: `cannot` instead of `can not`

Author: fixed

>   - Has Namespace Admin [permissions](#namespace-level-permissions) on all [Namespaces](/namespaces) in the account (can not be revoked)

Reviewer 2: ```suggestion
  - Has Namespace Admin [permissions](#namespace-level-permissions) on all [Namespaces](/namespaces) in the account. (Permissions cannot be revoked.)
```

- **Developer**
  - Can create Namespaces

>   - Is granted [Namespace Admin](/cloud/users#namespace-level-permissions) permission for each Namespace created (which can be revoked)

Reviewer 2: ```suggestion
  - Is granted [Namespace Admin](/cloud/users#namespace-level-permissions) permission for each Namespace created. This permission can be revoked
```
